server/test/users/users.controller.e2e-spec.ts

```diff
@@ -0,0 +1,155 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { userCreateDtoStub } from '../stubs';
+import { UsersController } from '../../src/users/users.controller';
+import { UsersService } from '../../src/users/users.service';
+import { User } from '../../src/users/entities/user.entity';
+import { UsersModule } from '../../src/users/users.module';
+import { getRepositoryToken, TypeOrmModule } from '@nestjs/typeorm';
+import { TEST_DB_OPTIONS } from '../testing-constants';
+import { INestApplication } from '@nestjs/common';
+import * as supertest from 'supertest';
+import { Repository } from 'typeorm';
+
+describe('UsersController', () => {
+  let app: INestApplication;
+  let usersService: UsersService;
+  let repository: Repository<User>;
+
+  let existingRecordId = 0;
+  const seed = {
+    first: 'peter',
+    last: 'parker',
+    email: 'peter.parker@example.com',
+    pass: 'secret1234',
+  };
+
+  beforeAll(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [UsersModule, TypeOrmModule.forRoot(TEST_DB_OPTIONS)],
+      controllers: [UsersController],
+      providers: [{ provide: getRepositoryToken(User), useClass: Repository }],
+    }).compile();
+
+    app = module.createNestApplication();
+    usersService = module.get<UsersService>(UsersService);
+    repository = module.get(getRepositoryToken(User));
+    // jest.clearAllMocks();
+    await app.init();
+  });
+
+  beforeEach(async () => {
+    // seed data
+    const { id } = await usersService.create({
+      first_name: seed.first,
+      last_name: seed.last,
+      email: seed.email,
+      password: seed.pass,
+    });
+    existingRecordId = id;
+  });
+
+  afterEach(async () => {
+    await repository.query(`DELETE FROM users;`);
+  });
```

Comment:
looks like now that some other stuff has landed there's a little more needed for this delete statement. got the below when i ran the tests a second time. and might be a good idea to merge in latest main to make sure nothing else like this is going on
```
postgres=# delete from users;
ERROR:  update or delete on table "users" violates foreign key constraint "FK_fe54927772536d8120e456a5994" on table "userOrganizations"
DETAIL:  Key (id)=(18) is still referenced from table "userOrganizations".
postgres=# 
```

Reply:
This error will not show up with new changes.